datafusion/core/tests/parquet/filter_pushdown.rs

```diff
@@ -223,7 +225,7 @@ async fn single_file() {
 
 #[tokio::test]
 async fn single_file_small_data_pages() {
-    let tempdir = TempDir::new().unwrap();
+    let tempdir = TempDir::new_in(Path::new(".")).unwrap();
```

Comment:
could this end up litterring in the local git checkout when running the tests?

Reply:
> could this end up litterring in the local git checkout when running the tests?

Emm, path which hold by `tempdir` will be delete after test case compeleted, Because struct `TempDir` implements `Drop` trait.
```rust
impl Drop for TempDir {
    fn drop(&mut self) {
        if !self.keep {
            let _ = remove_dir_all(self.path());
        }
    }
}
```

Reply:
> can you perhaps try to undo this change? (same above)

Actually, We must make this change. because I found out from the previous PR #13531 that the temp directory (i.g. "C:\\Users\\RUNNER\~1\\AppData\\Local\\Temp") that contains the special character `~` in CI, which will be replace to "%7E" when loading file, so that the file cannot be find. I notice that some parquet cases are working fine on windows machine (i.g. `parquet::file_statistics::check_stats_precision_with_filter_pushdown`). This cases read parquet files in "../../parquet-testing/data", which implies that the current folder definitely ok and does not have special characters (i.g. "~", "\\").
https://github.com/apache/datafusion/blob/fdb221fa645359b4bb65f600db1997ad6c865b65/datafusion/common/src/test_util.rs#L224-L229

Reply:
Thank you @zhuliquan  -- using `new_in` makes sense for me and I agree that the temporary file will be deleted when the `TempFile` goes out of scope

Reply:
> Thank you @zhuliquan -- using `new_in` makes sense for me and I agree that the temporary file will be deleted when the `TempFile` goes out of scope
Emm，using new_in is aim to make temp parquet file in current directory instead of system temp direction. It can avoid some special char (i.g ~). The feature which deleting file without scope is owe to `Drop` trait which is implemented by `TempDir`

Reply:
> the temporary file will be deleted when the TempFile goes out of scope

will this work too if a test panics?
tests creating files -- or _sometimes_ creating files -- in the git repo is something we shoulda void

Reply:
> > the temporary file will be deleted when the TempFile goes out of scope
> 
> will this work too if a test panics? tests creating files -- or _sometimes_ creating files -- in the git repo is something we shoulda void

Actually, `drop` will be invoked when test panic. If test happend panic, Rust will begin unwinding the stack, which means it will try to drop all variables that are still in scope (and therefore their drop methods will be called).